test/corelogging.jl

```diff
@@ -409,4 +409,8 @@ end
     @test record.group == :corelogging  # name of this file
 end
 
+@testset "complicated kwargs logging macro" begin
+    @test_logs (:warn, "foo")  @warn "foo" argvals=:((DoNotCare{$(Expr(:escape, :Any))}(),))
```

Comment:
The odd looking `$(Expr(:escape, :Any))` here could be confusing in future. I'd suggest something even simpler.

```suggestion
    @test_logs (:warn, "foo")  @warn "foo" argvals=:(some_nontrivial_quoted_expr(x))
```